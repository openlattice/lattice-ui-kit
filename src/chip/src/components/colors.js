import * as Colors from '../../../colors';

const {
  NEUTRAL,
  PURPLE,
  BLUE,
  TEAL,
  GREEN,
  YELLOW,
  ORANGE,
  RED,
  MAGENTA,
  VIOLET
} = Colors;

export default {
  default: {
    background: NEUTRAL.N00,
    color: NEUTRAL.N600
  },
  blue: {
    background: BLUE.B00,
    color: BLUE.B400,
    clickable: BLUE.B100
  },
  green: {
    background: GREEN.G00,
    color: GREEN.G400,
    clickable: GREEN.G100
  },
  magenta: {
    background: MAGENTA.M00,
    color: MAGENTA.M400,
    clickable: MAGENTA.M100
  },
  orange: {
    background: ORANGE.O00,
    color: ORANGE.O500,
    clickable: ORANGE.O100
  },
  purple: {
    background: PURPLE.P00,
    color: PURPLE.P400,
    clickable: PURPLE.P100
  },
  red: {
    background: RED.R00,
    color: RED.R400,
    clickable: RED.R100
  },
  teal: {
    background: TEAL.T00,
    color: TEAL.T500,
    clickable: TEAL.T100
  },
  violet: {
    background: VIOLET.V00,
    color: VIOLET.V400,
    clickable: VIOLET.V100
  },
  yellow: {
    background: YELLOW.Y00,
    color: YELLOW.Y500,
    clickable: YELLOW.Y100
  },
  primary: {
    color: 'white'
  },
  secondary: {
    color: PURPLE.P400
  },
};
