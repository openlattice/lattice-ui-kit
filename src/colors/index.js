/*
 * @flow
 */

/*
 * Inspiration:
 * https://atlassian.design/guidelines/brand/color-1
 * https://bitbucket.org/atlassian/atlaskit-mk-2/src/master/packages/core/theme
 * https://evergreen.surge.sh/components/colors
 */

import {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
} from './src/Colors';

export {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
};

export default {
  BLUE,
  GREEN,
  MAGENTA,
  NEUTRAL,
  ORANGE,
  PURPLE,
  RED,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
};
