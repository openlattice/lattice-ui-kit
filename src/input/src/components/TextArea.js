// @flow

import styled from 'styled-components';
import inputStyles from './styled/inputStyles';

const TextArea = styled.textarea`
  ${inputStyles}
  height: 100px;
  resize: vertical;
  width: 100%;
`;

export default TextArea;
