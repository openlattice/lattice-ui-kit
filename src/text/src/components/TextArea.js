// @flow

import styled from 'styled-components';
import inputStyles from './styled/inputStyles';

const TextArea = styled.textarea`
  height: 100px;
  resize: vertical;
  ${inputStyles}
`;

export default TextArea;
