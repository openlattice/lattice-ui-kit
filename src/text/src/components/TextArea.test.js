import React from 'react';

import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import TextArea from './TextArea';

import { RED } from '../../../colors';

describe('TextArea', () => {

  test('render matches snapshot', () => {
    const wrapper = mount(<TextArea />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('native input props', () => {
    test('should pass through any props to the input', () => {
      const nativeProps = {
        type: 'text',
        disabled: true,
        name: 'test',
        placeholder: 'test placeholder',
        maxLength: 8,
        min: 1,
        max: 8,
        required: true,
        autoComplete: 'on',
        form: 'test-form',
        pattern: '/.+/',
      };

      // eslint-disable-next-line react/jsx-props-no-spreading
      const wrapper = mount(<TextArea {...nativeProps} />)
        .find('textarea')
        .props();
      expect(wrapper).toEqual(expect.objectContaining(nativeProps));
    });

  });

  describe('Computed styles', () => {

    test('invalid', () => {
      const wrapper = mount(<TextArea invalid />);
      expect(wrapper).toHaveStyleRule('border', `1px solid ${RED.R300}`);
    });

  });

});
