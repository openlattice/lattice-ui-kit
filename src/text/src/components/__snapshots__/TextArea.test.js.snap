// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea render matches snapshot 1`] = `
.c0 {
  height: 100px;
  resize: vertical;
  background-color: white;
  border-radius: 3px;
  border: 1px solid #DFDFE3;
  box-shadow: none;
  box-sizing: border-box;
  color: #56565A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  font-size: 14px;
  min-height: 40px;
  padding: 10px;
  text-overflow: ellipsis;
  -webkit-transition: background-color 200ms ease-out,border-color 200ms ease-out;
  transition: background-color 200ms ease-out,border-color 200ms ease-out;
  width: 100%;
}

.c0:hover {
  background-color: #F2F2F4;
}

.c0:focus {
  border: solid 1px #56565A;
  background-color: white;
  outline: none;
}

.c0:disabled {
  background-color: #F2F2F4;
  color: #8B8B90;
  cursor: not-allowed;
}

<TextArea>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "TextArea-sc-10m8gjf-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "height:100px;resize:vertical;",
            "background-color:white;border-radius:3px;border:1px solid ",
            [Function],
            ";box-shadow:none;box-sizing:border-box;color:",
            "#56565A",
            ";display:flex;flex:0 1 auto;font-size:14px;min-height:40px;padding:10px;text-overflow:ellipsis;transition:background-color ",
            "200ms",
            " ease-out,border-color ",
            "200ms",
            " ease-out;width:",
            [Function],
            ";:hover{background-color:",
            "#F2F2F4",
            ";}:focus{border:solid 1px ",
            "#56565A",
            ";background-color:white;outline:none;}:disabled{background-color:",
            "#F2F2F4",
            ";color:",
            "#8B8B90",
            ";cursor:not-allowed;}",
          ],
        },
        "displayName": "TextArea",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "TextArea-sc-10m8gjf-0",
        "target": "textarea",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <textarea
      className="c0"
    />
  </StyledComponent>
</TextArea>
`;
